src/features/otsu_split.rs

```diff
@@ -0,0 +1,159 @@
+pub struct OtsuSplit {}
+
+lazy_info!(
+    OTSU_SPLIT_INFO,
+    OtsuSplit,
+    size: 4,
+    min_ts_length: 2,
+    t_required: false,
+    m_required: true,
+    w_required: false,
+    sorting_required: false,
+);
+
+impl OtsuSplit {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn doc() -> &'static str {
+        DOC
+    }
+}
+
+impl FeatureNamesDescriptionsTrait for OtsuSplit {
+    fn get_names(&self) -> Vec<&str> {
+        vec![
+            "otsu_mean_diff",
+            "otsu_std_lower",
+            "otsu_std_upper",
+            "otsu_lower_to_all_ratio",
+        ]
+    }
+
+    fn get_descriptions(&self) -> Vec<&str> {
+        vec!["difference between mean values of Otsu split subsets",
+             "standard deviation for observations below the threshold given by Otsu method",
+             "standard deviation for observations above the threshold given by Otsu method",
+             "ratio of quantity of observations bellow the threshold given by Otsu method to quantity of all observations"]
+    }
+}
+
+impl<T> FeatureEvaluator<T> for OtsuSplit
+where
+    T: Float,
+{
+    fn eval(&self, ts: &mut TimeSeries<T>) -> Result<Vec<T>, EvaluatorError> {
+        self.check_ts_length(ts)?;
+
+        if ts.m.get_max() == ts.m.get_min() {
+            return Err(EvaluatorError::FlatTimeSeries);
+        }
+
+        let mut dm = T::zero();
+        let mut w: usize = 0;
+        let mean = ts.m.get_mean();
+        let count = ts.lenu();
+        let msorted = ts.m.get_sorted();
+        let mut last_variance = T::zero();
+
+        for &m in msorted.iter() {
+            w += 1;
+            dm += mean - m;
+
+            if w == count {
+                break;
+            }
+
+            let variance =
+                dm * dm / (count - w).value_as::<T>().unwrap() / w.value_as::<T>().unwrap();
+
+            if variance < last_variance {
+                break;
+            }
+            last_variance = variance;
+        }
+
+        let std_lower;
+        let std_upper;
+        let mean_lower;
+        let mean_upper;
+
+        if w == 2 {
+            std_lower = T::zero();
+            mean_lower = msorted[0];
+        } else {
+            let mut lower: DataSample<_> = msorted[0..w - 1].into();
+            std_lower = lower.get_std();
+            mean_lower = lower.get_mean()
+        }
+
+        if (count - w) == 0 {
+            std_upper = T::zero();
+            mean_upper = msorted[count - 1];
+        } else {
+            let mut upper: DataSample<_> = msorted[w..count].into();
+            std_upper = upper.get_std();
+            mean_upper = upper.get_mean();
+        }
+
+        // let mean_diff = upper.get_mean() - lower.get_mean();
+        let mean_diff = mean_upper - mean_lower;
+        let lower_to_all = (w - 1).value_as::<T>().unwrap() / count.value_as::<T>().unwrap();
+
+        Ok(vec![mean_diff, std_lower, std_upper, lower_to_all])
+    }
+}
+
+#[cfg(test)]
```

Comment:
Please test more edge-cases: 1) minimum number of observations, 2) `w == count` (if you could find such an example for non-flat time-series), 3) w == 2